src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Conversion.cs

```diff
@@ -307,6 +307,15 @@ private static bool IsFloatPointExpressionOfUnknownPrecision(BoundExpression rew
 
                     break;
 
+                case ConversionKind.MethodGroup:
+
+                    if (!symbolOpt.IsStatic || isExtensionMethod || _inExpressionLambda || _factory.TopLevelMethod.MethodKind == MethodKind.StaticConstructor)
+                    {
+                        break;
```

Comment:
It is not obvious why do we have to break for each condition in the ```if````. Please add a comment explaining reasons behind each condition.